zipline/testing/predicates.py

```diff
@@ -560,6 +567,18 @@ def assert_array_equal(result,
             err_msg=msg,
         )
     except AssertionError as e:
+        if is_datetime:
```

Comment:
Hmm. this feels kinda gross. I feel like it might be cleaner to just delegate to an `assert_datetime_array_equal` method so that we don't have to keep the flag around.

Reply:
(I think this comment is also moot if we use `assert_array_compare`)